js-envs/src/main/scala/org/scalajs/jsenv/AsyncJSRunner.scala

```diff
@@ -38,12 +38,25 @@ trait AsyncJSRunner {
 
   /** Await completion of the started Run for the duration specified
    *  by [[atMost]]. Strictly equivalent to:
-   * 
+   *
    *  {{{
    *  Await.result(future, atMost)
    *  }}}
-   * 
+   *
    */
   final def await(atMost: Duration): Unit = Await.result(future, atMost)
 
+  /** Await completion of the started Run for the duration specified by
+   *  [[atMost]], or force it to stop if the timeout expires.
+   */
+  final def awaitOrStop(atMost: Duration): Unit = {
+    try {
+      await(atMost)
+    } catch {
+      case th: Throwable =>
```

Comment:
It doesn't really matter, though, does it? Won't `stop` be a no-op if the vm has already stopped? In that case, the exception is properly thrown again. And if there is an exception that *doesn't* stop it, well the `stop` will stop it anyway, and still throw the right exception.